packages/react-instantsearch-core/src/lib/voiceSearchHelper/index.ts

```diff
@@ -0,0 +1,131 @@
+const STATUS_INITIAL = 'initial';
+const STATUS_ASKING_PERMISSION = 'askingPermission';
+const STATUS_WAITING = 'waiting';
+const STATUS_RECOGNIZING = 'recognizing';
+const STATUS_FINISHED = 'finished';
+const STATUS_ERROR = 'error';
+
+type QueryChangeListener = (query: string) => void;
+type StateChangeListener = () => void;
+
+export type VoiceListeningState = {
+  status: string;
+  transcript?: string;
+  isSpeechFinal?: boolean;
+  errorCode?: string;
+};
+
+export type ToggleListening = () => void;
+
+export default function voiceSearchHelper() {
+  const SpeechRecognitionAPI: new () => SpeechRecognition =
+    (window as any).webkitSpeechRecognition ||
+    (window as any).SpeechRecognition;
+  const getDefaultState = (status: string): VoiceListeningState => ({
+    status,
+    transcript: undefined,
+    isSpeechFinal: undefined,
+    errorCode: undefined,
+  });
+  let state: VoiceListeningState = getDefaultState(STATUS_INITIAL);
+  let recognition: SpeechRecognition | undefined;
+  let searchAsYouSpeak;
+  // tslint:disable-next-line:no-empty
```

Comment:
This rule should be disabled at the project level: https://github.com/algolia/react-instantsearch/blob/a94e73a53dbefab5383cd67f2d959d75c51a3eb2/tslint.json#L15

Reply:
Good. I've merged master to this branch and now I have it :)